test/ft/di_config.cpp

```diff
@@ -68,6 +68,36 @@ test constructible_policy_must_be_bound = [] {
   injector.create<c>();
 };
 
+template <class>
+struct is_shared_ptr : std::false_type {};
+
+template <class T>
+struct is_shared_ptr<std::shared_ptr<T>> : std::true_type {};
+
+test constructible_policy_creatable_type = [] {
+  class must_be_creatable_or_bound : public di::config {
+   public:
+    static auto policies(...) noexcept {
+      using namespace di::policies;
+      using namespace di::policies::operators;
+
+      const auto is_bound_root_shared_ptr = is_root<di::_>{} && is_shared_ptr<di::_>{} && is_bound<di::_>{};
+      const auto is_bound_not_root = !is_root<di::_>{} && is_bound<di::_>{};
+
+      return di::make_policies(constructible<true>(is_bound_root_shared_ptr || is_bound_not_root));
```

Comment:
@krzysztof-jusiak Being hones I don't think this `<true>` is self-explained. It would be better to have anything like:
`constructible_including_root` and `constructible_excluding_root`. I didn't noticed this `<true>` first time and was quite surprised that `is_root` doesn't work

Reply:
Yeah, I started with that but couldn't find a good name ;) so I let the user decide. It's a limitation of the C++ since we can't do IncludeRoot = true, but user can specify any name they want and just use it instead of true/false

Reply:
@krzysztof-jusiak What about having just argument is_root_included? It might be more explicit though